### PR TITLE
Documentation and changes to installation/maintenance infrastructure

### PR DESCRIPTION
**Pull request by arokem**
For now, I have only started working on the infrastructure side of things, primarily simplifying the setup file in favor of more information in the setup.cfg and (newly added) pyproject.toml file. 

Next: 

- [x] Add sphinx documentation.
- [x] Add a documentation build workflow.

### CODE REVIEW

**Comment by arokem** [comment]
This is ready for review. The GitHub Actions workflow is configured so that the website gets updated every time that we push a tag to the repository (e.g., for a release). This means that once this is merged, we won't have a website until we push a tag. I guess we've already been talking about cutting a first release over on #33, so that will give us another reason to do so. 

The content of the webpage so far is pretty rudimentary and based primarily on the OHBM abstract. If you are interested in commenting on the content, you can find it [here](https://github.com/tee-ar-ex/trx-python/pull/36/files#diff-4eec6cec5f6fab1548b85433ea8ca81315ae165db4b7f84019f287df9015699f). 

If you want to download the content to your machine to see how this gets rendered, you can grab the workflow build artifact [here](https://github.com/tee-ar-ex/trx-python/suites/7085466567/artifacts/279997242). Unzip the file and open the html index file and you can view the website in your browser. This sphinx theme can be [customized a lot!](https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/configuring.html)

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'trx-python'
+copyright = '2022, The TRX developers'
```
Comment (skoudoro) [inline docs/source/conf.py:21]: To reduce the maintenance, you can use:
`copyright = '2021-{0}, The TRX developers'.format(datetime.now().year)`

or f-string

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            # The type of image to be used (see below for details)
+            "type": "fontawesome",
+        }
+   ]
+}
+
+
+autoapi_type = 'python'
+autoapi_dirs = ['../../trx']
+autoapi_ignore = ['*test*', '*version*', 'License']
```
Comment (skoudoro) [inline docs/source/conf.py:86]: what is it? Curiosity

**Review by skoudoro — COMMENTED** [review]
LGTM!

I confirm that this theme is great! I love it! we use it for [fury.gl](https://fury.gl) and I would like to start using it for DIPY to simplify the current craziness

File: docs/source/conf.py (hunk 2 above)
Comment (arokem) [reply docs/source/conf.py:86]: IIUC, if you don't ignore this, you'll get API documentation for your test and version files. I can probably remove the "License" entry, though.

File: docs/source/conf.py (hunk 1 above)
Comment (arokem) [reply docs/source/conf.py:21]: Nice! Automate everything.

**Comment by arokem** [comment]
Thanks for taking a look and for the suggestions. Implemented in 7de747e

**Comment by arokem** [comment]
Hey @frheault : I saw that you're back from your travels. Any thoughts about this? Might be good to merge the infra changes before proceeding with other developments

**Comment by frheault** [comment]
@arokem I like it! This should be merged before #33 !

Then I will try to re-upload all data with the N+1 'correction' on Figshare with version 0.1 and then we should try to do a release candidate with the HTML, the N+1 'fix' and a working dataset on Figshare.

PS: 0.1 should be a tag on every trx repo if they work with the data on FigShare. I think that if we update the specification we could switch to 1.*, then 2.* so we can follow along dataset and code together

**Comment by arokem** [comment]
Sounds good! Looks like I need to rebase this on top of master before it can be merged. Let me do that real quick.

**Comment by arokem** [comment]
Now rebased on master. Note in particular that I've slimmed down `setup.py` quite significantly. One thing that I was a little bit averse to is the control of installation through setting of environmental variables. I think that we can configure different installation options in `setup.cfg` instead.

**Comment by arokem** [comment]
Looks like there are still some installation issues here. I don't have time to figure this out today, so will have to look at again, maybe maybe maybe later this week.

**Comment by arokem** [comment]
Sorry for the long gap here. Looks like it's installing and testing fine now. 

@frheault : do you have the bandwidth to take another look and maybe merge this? 

Of note, when testing, DIPY and FURY are now installed from their release version. Is that going to be a problem? I can try fixing this on a subsequent PR, but would like to get the documentation online quickly, by merging this as is, if that's possible.

File: setup.py [hunk 3]
```diff
@@ -1,25 +1,6 @@
...
-                'dipy@git+https://github.com/frheault/dipy.git@4e192c5c6#egg=dipy',
-                'pytest>=7.*', 'pytest-console-scripts>=0.*']
+from setuptools import setup
 
 setup(name='trx',
-      packages=find_packages(),
-      setup_requires=['packaging>=19.0'],
-      install_requires=REQUIRES,
-      scripts=SCRIPTS)
+      use_scm_version=True,
```
Comment (skoudoro) [inline setup.py:5]: Curiosity:  This can not be added to `setup.cfg` or `pyproject.toml` ?

Comment (arokem) [reply setup.py]: Possibly. See [04ead56](https://github.com/tee-ar-ex/trx-python/pull/36/commits/04ead56c98dc7107aef6777b6ad3dd22c90fb434). 

However, I believe that we'd still need the setup.py file to flexibly install the scripts using the glob pattern.

**Comment by skoudoro** [comment]
I think this PR is important to be merged as soon as possible. 

Currently, when you create a new environment, `trx-pytho`n failed to install because of missing libraries like `packaging`. Also, it is strange to have a version named 0.0.0.

@frheault, Can you have a closer look on this PR and go ahead to merge it. Thanks for your future feedback

**Comment by frheault** [comment]
I have a fresh Ubuntu 22.04 installed and this PR made the installation very easy, LGTM!

**Pull request merged**